src/qpmodel.jl

```diff
@@ -42,11 +42,11 @@ function QuadraticModel(
   Hvals::AbstractVector;
   Arows::AbstractVector{<:Integer} = Int[],
   Acols::AbstractVector{<:Integer} = Int[],
-  Avals::AbstractVector = T[],
-  lcon::AbstractVector = T[],
-  ucon::AbstractVector = T[],
-  lvar::AbstractVector = fill(T(-Inf), length(c)),
-  uvar::AbstractVector = fill(T(Inf), length(c)),
+  Avals::AbstractVector = similar(c, 0),
+  lcon::AbstractVector = similar(c, 0),
+  ucon::AbstractVector = similar(c, 0),
+  lvar::AbstractVector = fill!(similar(c, length(c)), T(-Inf)),
+  uvar::AbstractVector = fill!(similar(c, length(c)), T(Inf)),
   c0::T = zero(T),
   kwargs...,
 ) where {T}
```

Comment:
You should change these to `::S` and use `T = eltype(S)`.

Reply:
OK, thanks! You mean I should do this?
```
function QuadraticModel(
  c::S,
  Hrows::AbstractVector{<:Integer},
  Hcols::AbstractVector{<:Integer},
  Hvals::S;
  Arows::AbstractVector{<:Integer} = Int[],
  Acols::AbstractVector{<:Integer} = Int[],
  Avals::AbstractVector = S(undef, 0),
  lcon::AbstractVector = S(undef, 0),
  ucon::AbstractVector = S(undef, 0),
  lvar::AbstractVector = fill!(S(undef, length(c)), eltype(c)(-Inf)),
  uvar::AbstractVector = fill!(S(undef, length(c)), eltype(c)(Inf)),
  c0::T = zero(T),
  kwargs...,
) where {T}
```

Reply:
Yes, but also `lcon`, `ucon`, etc.. Because `NLPModelMeta` will require them to be type `S`. Probably `Avals` should be `::S` too.